pkg/logs/tail_test.go

```diff
@@ -0,0 +1,323 @@
+package logs
+
+import (
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+const responseFile = "testdata/response.json"
+
+func TestValidateTimes(t *testing.T) {
+	dur, _ := time.ParseDuration("10s")
+	for i, test := range []struct {
+		in  cfg
+		exp cfg
+	}{
+		{
+			in:  cfg{from: 1601480668, to: 1601480768, searchPadding: dur},
+			exp: cfg{from: 1601480658, to: 1601480778, searchPadding: dur},
+		},
+		{
+			in:  cfg{from: 1601480668, to: 1601480768},
+			exp: cfg{from: 1601480668, to: 1601480768},
+		},
+		{
+			in:  cfg{searchPadding: dur},
+			exp: cfg{searchPadding: dur},
+		},
+	} {
+		c := TailCommand{cfg: test.in}
+		c.adjustTimes()
+		if equal := reflect.DeepEqual(test.exp, c.cfg); !equal {
+			t.Errorf("#%d: validateTimes mismatch: got: %#+v  want: %#+v", i, c.cfg, test.exp)
+		}
+	}
+}
+
+func TestSplitByReqID(t *testing.T) {
+	full := []Log{
+		{SequenceNum: 1, RequestID: "41f82900"},
+		{SequenceNum: 2, RequestID: "41f82900"},
+		{SequenceNum: 3, RequestID: "2bef4613"},
+		{SequenceNum: 4, RequestID: "2bef4613"},
+		{SequenceNum: 5, RequestID: "41f82900"},
+		{SequenceNum: 6, RequestID: "41f82900"},
+		{SequenceNum: 6, RequestID: "2bef4613"},
+		{SequenceNum: 1, RequestID: "2bef4613"},
+		{SequenceNum: 5, RequestID: "2bef4613"},
+		{SequenceNum: 2, RequestID: "2bef4613"},
+		{SequenceNum: 3, RequestID: "41f82900"},
+		{SequenceNum: 4, RequestID: "41f82900"},
+	}
+
+	expfull := map[string][]Log{
+		"41f82900": {
+			{SequenceNum: 1, RequestID: "41f82900"},
+			{SequenceNum: 2, RequestID: "41f82900"},
+			{SequenceNum: 5, RequestID: "41f82900"},
+			{SequenceNum: 6, RequestID: "41f82900"},
+			{SequenceNum: 3, RequestID: "41f82900"},
+			{SequenceNum: 4, RequestID: "41f82900"},
+		},
+		"2bef4613": {
+			{SequenceNum: 3, RequestID: "2bef4613"},
+			{SequenceNum: 4, RequestID: "2bef4613"},
+			{SequenceNum: 6, RequestID: "2bef4613"},
+			{SequenceNum: 1, RequestID: "2bef4613"},
+			{SequenceNum: 5, RequestID: "2bef4613"},
+			{SequenceNum: 2, RequestID: "2bef4613"},
+		},
+	}
+
+	single := []Log{
+		{SequenceNum: 1, RequestID: "41f82900"},
+	}
+
+	expsingle := map[string][]Log{
+		"41f82900": {{SequenceNum: 1, RequestID: "41f82900"}},
+	}
+
+	for i, test := range []struct {
+		in   []Log
+		want map[string][]Log
+	}{
+		{in: full, want: expfull},
+		{in: single, want: expsingle},
+		{in: []Log{}, want: make(map[string][]Log)},
+	} {
+		got := splitByReqID(test.in)
+		if diff := cmp.Diff(test.want, got); diff != "" {
+			t.Errorf("#%d: parseResponseData mismatch (-want +got):\n%s", i, diff)
```

Comment:
I believe this should be what's referenced as that's the function actually called.
```suggestion
			t.Errorf("#%d: splitByReqID mismatch (-want +got):\n%s", i, diff)
```

Reply:
Yes, and will add a comment to the test function.